scripts/zones/Dynamis-Xarcabard/mobs/Vanguard_Eye.lua

```diff
@@ -35,27 +35,47 @@ end;
 -----------------------------------
 
 function onMobDeath(mob,killer)
-	
-	local mobID = mob:getID();
-	
-	-- 035 039: spawn 043 when defeated
-	if (mobID == 17326536) then
-		SpawnMob(17326553);
-	-- 058: spawn 60 when defeated
-	elseif (mobID == 17326661 and GetMobAction(17326668) == 0 and GetMobAction(17326673) == 0 or 
-		   mobID == 17326668 and GetMobAction(17326661) == 0 and GetMobAction(17326673) == 0 or 
-		   mobID == 17326673 and GetMobAction(17326661) == 0 and GetMobAction(17326668) == 0) then
-		SpawnMob(17326706);
-	-- 114: spawn 112 when defeated
-	elseif (mobID == 17326790) then
-		SpawnMob(17326086);
-	-- 144-149: spawn 150 when defeated
-	elseif (mobID >= 17330913 and mobID <= 17330918) then
-		SetServerVariable("[DynaXarcabard]TE150_Trigger",GetServerVariable("[DynaXarcabard]TE150_Trigger") + (mobID - 17330912) ^ 2);
-	end
-	
-	if (GetServerVariable("[DynaXarcabard]TE150_Trigger") == 63) then
-		SpawnMob(17330919); -- 150
-	end
-	
-end;
+    
+    local mobID = mob:getID();
+    
+    -- 035 039: spawn 043 when defeated
+    if (mobID == 17330718) then
+        local southTE = GetServerVariable("[DynaXarcabard]TE43_Trigger");
+        if (bit.band(southTE, bit.lshift(1,0)) ~= 1) then
+            SetServerVariable("[DynaXarcabard]TE43_Trigger",bit.bor(southTE, bit.lshift(1,0)))
+        end
+    elseif (mobID == 17330756)  then
+        local southTE = GetServerVariable("[DynaXarcabard]TE43_Trigger");
+        if (bit.band(southTE, bit.lshift(1,1)) ~= 1) then

```

Comment:
this statement can't ever be false - the band is going to either be 2 or 0.  If you're checking whether a bit is set or net, compare it with 0 (you either have compare with 0 or with the shifted number again, and 0 is easier)